### PR TITLE
Support assigned user roles on private invite links

### PR DESCRIPTION
**Pull request by Devorein**
This PR allows us to assign roles to private links, just like we do for token gates. You can select the roles that you want to assign, and when any user joins the workspace via that invite link they will automatically get assigned those roles.

### CODE REVIEW

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx [hunk 1]
```diff
@@ -11,19 +11,30 @@ import Typography from '@mui/material/Typography';
...
 import UserDisplay from 'components/common/UserDisplay';
+import TokenGateRolesSelect from 'components/settings/roles/components/TokenGates/components/TokenGateRolesSelect';
+import { useCurrentSpace } from 'hooks/useCurrentSpace';
 import type { InviteLinkPopulated } from 'pages/api/invites/index';
 
 interface Props {
   isAdmin: boolean;
   invites: InviteLinkPopulated[];
   onDelete: (invite: InviteLinkPopulated) => void;
+  refetchInvites: KeyedMutator<InviteLinkPopulated[]>;
```
Comment (msadura) [inline components/settings/contributors/InviteLinks/InviteLinksTable.tsx:28]: nice you are just invoking `refetchInvites()` type can be as simple as `() => void` or use built in `VoidFunction` type https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.voidfunction.html

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx [hunk 2]
```diff
@@ -32,6 +43,24 @@ export default function InvitesTable (props: Props) {
...
+    }
+  }
+
+  async function deleteRoleFromInviteLink (inviteLinkId: string, roleId: string) {
+    const inviteLink = invites.find(_tokenGate => _tokenGate.id === inviteLinkId);
+    if (inviteLink && space) {
+      const roleIds = inviteLink.inviteLinkToRoles
+        .map(inviteLinkToRole => inviteLinkToRole.roleId)
+        .filter(inviteLinkRoleId => inviteLinkRoleId !== roleId);
+      await charmClient.updateInviteLinkRoles(inviteLinkId, space.id, roleIds);
```
Comment (msadura) [inline components/settings/contributors/InviteLinks/InviteLinksTable.tsx:59]: you can reuse `updateInviteLinkRoles` function defined above

**Review by msadura — APPROVED** [review]
Just 2 small things and should be good to go :)

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx [hunk 3]
```diff
@@ -32,6 +43,24 @@ export default function InvitesTable (props: Props) {
...
 
+  async function updateInviteLinkRoles (inviteLinkId: string, roleIds: string[]) {
+    if (space) {
+      await charmClient.updateInviteLinkRoles(inviteLinkId, space.id, roleIds);
+      refetchInvites();
+    }
+  }
+
+  async function deleteRoleFromInviteLink (inviteLinkId: string, roleId: string) {
+    const inviteLink = invites.find(_tokenGate => _tokenGate.id === inviteLinkId);
```
Comment (msadura) [inline components/settings/contributors/InviteLinks/InviteLinksTable.tsx:54]: just a cosmetic thing - as far as i understand it is not `tokenGate` but `inviteLink` ? Name is a bit misleading

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx (hunk 1 above)
Comment (Devorein) [reply components/settings/contributors/InviteLinks/InviteLinksTable.tsx:28]: Nice one. Never knew about `VoidFunction` before, added to my toolbelt

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx (hunk 2 above)
Comment (Devorein) [reply components/settings/contributors/InviteLinks/InviteLinksTable.tsx:59]: Updated. Thanks.

File: components/settings/contributors/InviteLinks/InviteLinksTable.tsx (hunk 3 above)
Comment (Devorein) [reply components/settings/contributors/InviteLinks/InviteLinksTable.tsx:54]: Oops. Copypasta code smell 😝

**Pull request merged**